lib/scnnr/routing.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Scnnr
+  class Routing
+    API_SCHEME = URI::HTTPS
+    API_HOST = 'api.scnnr.cubki.jp'
+
+    attr_reader :path_prefix
+
+    def initialize(path, path_prefix, params, allowed_params)
+      @path = path
+      @path_prefix = path_prefix
+      @queries = build_queries params, allowed_params
+    end
+
+    def to_url
+      API_SCHEME.build(
+        host: API_HOST,
+        path: self.path,
+        query: query_string
+      )
+    end
+
+    def queries
+      @queries.reject { |_, val| val.nil? }
```

Comment:
Ah, I want to use `Hash#compact`, but Ruby 2.3.x does not support it!
By the way we may want to support Ruby 2.5 as well...😱

Reply:
Yes, it's so inconvenient...
But I expect that supporting Ruby 2.5 is not big deal. Let's do that in the near future!